iceoryx2-bb/posix/tests/process_state_tests.rs

```diff
@@ -0,0 +1,302 @@
+    file.set_permission(Permission::OWNER_ALL).unwrap();
+    file.remove_self().unwrap();
+    assert_that!(monitor.state().unwrap(), eq ProcessState::DoesNotExist);
+}
+
+#[test]
+pub fn process_state_cleaner_cannot_be_created_when_process_does_not_exist() {
+    let path = generate_file_path();
+    let mut cleaner_path = path.clone();
+    cleaner_path.push_bytes(b"_cleanup").unwrap();
+
+    let cleaner = ProcessCleaner::new(&path);
+    assert_that!(cleaner, is_err);
+    assert_that!(
+        cleaner.err().unwrap(), eq
+        ProcessCleanerCreateError::DoesNotExist
+    );
+
+    let file = FileBuilder::new(&path)
+        .has_ownership(true)
+        .creation_mode(CreationMode::PurgeAndCreate)
+        .create()
+        .unwrap();
+
+    let cleaner = ProcessCleaner::new(&path);
+    assert_that!(cleaner, is_err);
+    assert_that!(
+        cleaner.err().unwrap(), eq
+        ProcessCleanerCreateError::DoesNotExist
+    );
+    drop(file);
+    std::thread::sleep(Duration::from_millis(100));
+
+    let _file = FileBuilder::new(&cleaner_path)
+        .has_ownership(true)
+        .creation_mode(CreationMode::PurgeAndCreate)
+        .create()
+        .unwrap();
+
+    let cleaner = ProcessCleaner::new(&path);
+    assert_that!(cleaner, is_err);
+    assert_that!(
+        cleaner.err().unwrap(), eq
+        ProcessCleanerCreateError::DoesNotExist
+    );
+}
+
+// START: OS with IPC only lock detection
+//
+// the lock detection does work on some OS only in the inter process context.
+// In the process local context the lock is not detected when the fcntl GETLK call is originating
+// from the same thread os the fcntl SETLK call. If it is called from a different thread GETLK
+// blocks despite it should be non-blocking.
+#[test]
+#[cfg(not(any(target_os = "linux", target_os = "freebsd", target_os = "macos")))]
+pub fn process_state_monitor_detects_alive_state_from_existing_process() {
+    let path = generate_file_path();
+
+    let guard = ProcessGuard::new(&path).unwrap();
+    let monitor = ProcessMonitor::new(&path).unwrap();
+
+    assert_that!(monitor.state().unwrap(), eq ProcessState::Alive);
+    drop(guard);
+    assert_that!(monitor.state().unwrap(), eq ProcessState::DoesNotExist);
+}
+
+#[test]
+#[cfg(not(any(target_os = "linux", target_os = "freebsd", target_os = "macos")))]
+pub fn process_state_cleaner_cannot_be_acquired_from_living_process() {
+    let path = generate_file_path();
+
+    let _guard = ProcessGuard::new(&path).unwrap();
+    let cleaner = ProcessCleaner::new(&path);
+    assert_that!(cleaner, is_err);
+    assert_that!(
+        cleaner.err().unwrap(), eq
+        ProcessCleanerCreateError::ProcessIsStillAlive
+    );
+}
+
+#[test]
+#[cfg(not(any(target_os = "linux", target_os = "freebsd", target_os = "macos")))]
+pub fn process_state_cleaner_cannot_be_acquired_twice() {
+    let path = generate_file_path();
+    let mut cleaner_path = path.clone();
+    cleaner_path.push_bytes(b"_cleanup").unwrap();
```

Comment:
We need to implement the error trait for all errors, then we can use `?` in the tests.

Reply:
Yes, but I would keep it for later. When the repo is too clean right from the beginning, people are maybe intimidated by our perfectness ;)